distributed/tests/test_worker.py

```diff
@@ -3363,7 +3363,9 @@ async def gather_dep(self, *args, **kwargs):
 
 
 @pytest.mark.parametrize("intermediate_state", ["resumed", "cancelled"])
-@pytest.mark.parametrize("close_worker", [False, True])
+@pytest.mark.parametrize(
+    "close_worker", [False, pytest.param(True, marks=pytest.mark.slow)]
+)
```

Comment:
(cancelled, True) now takes 5s instead of 100ms as now the network comms is fired blindly.
(resumed, True) was already taking 5s before this PR.

Reply:
Is that related to https://github.com/dask/distributed/issues/6354 at all? Because `Scheduler.remove_worker` doesn't flush or await the `BatchedSend`, so after `remove_worker` returns, there's still some delay until it receives the message and actually shuts down? 5s seems longer than I'd expect.

Reply:
I'm getting this traceback:
```
  File "/home/crusaderky/github/distributed/distributed/worker.py", line 4575, in _get_data
    comm = await rpc.connect(worker)
  File "/home/crusaderky/github/distributed/distributed/core.py", line 1184, in connect
    return await connect_attempt
  File "/home/crusaderky/github/distributed/distributed/core.py", line 1120, in _connect
    comm = await connect(
  File "/home/crusaderky/github/distributed/distributed/comm/core.py", line 315, in connect
    raise OSError(
OSError: Timed out trying to connect to tcp://127.0.0.1:34011 after 5 s
```
There seems to be nothing, when Worker.close() is invoked, that explicitly shuts down the RPC channel.

Reply:
https://github.com/dask/distributed/blob/33fc50ca9817216bb4105b68f5e0859ebfb80fdb/distributed/worker.py#L1568
maybe this just happens too late?

Reply:
Moving to https://github.com/dask/distributed/issues/6409